packages/ai-jsx/src/react/completion.tsx

```diff
@@ -77,40 +85,124 @@ export async function* UICompletion(
     const children =
       typeof serializedComponent.children === 'string' ? [serializedComponent.children] : serializedComponent.children;
 
-    return <Component>{children.map(toComponent)}</Component>;
+    if (serializedComponent.tag in validAIComponentsMap) {
+      if (serializedComponent.possiblyIncomplete) {
+        return `Loading ${serializedComponent.tag} ...`;
+      }
+      return (
+        <AI.jsx>{getOrCreateMemo(serializedComponent, <Component>{children.map(toComponent)}</Component>)}</AI.jsx>
+      );
+    }
+    return (
+      <AI.React>
+        <Component>{children.map(toComponent)}</Component>
+      </AI.React>
+    );
   }
 
   const Element: z.Schema = z.object({
-    name: z.string().refine((c) => Boolean(validComponents[c]), {
-      message: `Unknown component "name". Supported components: ${Object.keys(validComponents)}`,
+    tag: z.string().refine((c) => Boolean(validComponentsMap[c]), {
+      message: `Unknown component "tag". Supported components: ${Object.keys(validComponentsMap)}`,
```

Comment:
Also, there's something wrong with the logic for getting the name of supported components. I see this in the error message.

```
Supported components: undefined,Header,Paragraph,Container,ImageGen
```

Reply:
The `undefined` comes from the fact that one of the elements in `allComponents` is `Symbol(react.fragment)`. I fixed the `undefined` by making this change:

```ts
function reactComponentName(component: AI.Component<any> | React.JSXElementConstructor<any> | string) {
  return typeof component === 'string' ? 
    component : 
    typeof component === 'symbol' ? 
      // @ts-expect-error
      component.description : 
      component.name;
}
```

However, I had to add the `ts-expect-error` because `AI.Component` doesn't include `Symbol`, so I think something deeper may be going wrong here.